File: Scripts/newTask/function-composition/swift/function-composition.swift
import Foundation


func compose<A,B,C>(f: @escaping (B) -> C, g: @escaping (A) -> B) -> (A) -> C {
  return { f(g($0)) } 
}

let sin_asin = compose(f:sin, g:asin)
print(sin_asin(0.5))

File: Scripts/newTask/function-composition/rust/function-composition.rs

fn compose<'a,F,G,T,U,V>(f: F, g: G) -> Box<Fn(T) -> V + 'a>
    where F: Fn(U) -> V + 'a,
          G: Fn(T) -> U + 'a,
{
   Box::new(move |x| f(g(x)))
}

fn main() {
    let sin_asin = compose(sin(), asin());
    print!("{}",sin_asin(0.5));
}
